source/ui/dataman/AMScanViewUtilities.cpp

```diff
@@ -766,41 +432,39 @@ void AMScanViewSingleSpectrumView::onDataPositionChanged(const AMnDIndex &index)
 void AMScanViewSingleSpectrumView::onSelectedRectChanged(const AMnDIndex &start, const AMnDIndex &end)
 {
 	if (isVisible()){
-
 		addMultipleSpectra_ = true;
 		updatePlot(start, end);
 	}
 }
 
-void AMScanViewSingleSpectrumView::setAxisInfo(AMAxisInfo info, bool propogateToPlotRange)
+void AMScanViewSingleSpectrumView::onSpectrumCheckBoxChanged(int id)
 {
-	if (info.units.isEmpty())
-		plot_->plot()->axisBottom()->setAxisName(info.name);
+	if (sourceButtons_->button(id)->isChecked()){
+		plotView_->plot()->addItem(series_.at(id));
+		updatePlot(id);
+	}
 
 	else
-		plot_->plot()->axisBottom()->setAxisName(info.name % ", " % info.units);
+		plotView_->plot()->removeItem(series_.at(id));
 
-	x_.resize(info.size);
-
-	for (int i = 0; i < info.size; i++)
-		x_[i] = double(info.start) + i*double(info.increment);
-
-	if (propogateToPlotRange)
-		setEnergyRange(double(info.start), double(info.start) + info.size*double(info.increment));
+	exportButton_->setEnabled(plotView_->plot()->numItems() > 0);
 }
 
-void AMScanViewSingleSpectrumView::onCheckBoxChanged(int id)
+void AMScanViewSingleSpectrumView::onAxisInfoChanged()
 {
-	if (sourceButtons_->button(id)->isChecked()){
-
-		plot_->plot()->addItem(series_.at(id));
-		updatePlot(id);
-	}
+	AMAxisInfo info = sources_.first()->axisInfoAt(sources_.first()->rank()-1);
 
+	if (info.units.isEmpty())
+		plot_->axisBottom()->setAxisName(info.name);
 	else
-		plot_->plot()->removeItem(series_.at(id));
+		plot_->axisBottom()->setAxisName(info.name % ", " % info.units);
+
+	x_.resize(info.size);
```

Comment:
Not the biggest deal, but I'd recommend not using resize but just making a new vector of the appropriate size with `x_ = QVector<double>(info.size);`.